### PR TITLE
[Relay] Finish implementations of WithFields

### PR DESCRIPTION
**Pull request by mbs-octoml**
This has been sitting in my collage branch forever, might as well clear it into main.

I added the beginnings of a unit test, but only for the Call overload.

### CODE REVIEW

File: src/relay/ir/expr.cc [hunk 1]
```diff
@@ -27,6 +27,26 @@
 
 namespace tvm {
 
+GlobalVar WithFields(GlobalVar global_var, Optional<String> opt_name_hint, Optional<Type> opt_type,
```
Comment (Mousius) [inline src/relay/ir/expr.cc:30]: Could you add some tests for `GlobalVar` and `Constant` variants of `WithFields`?

**Review by Mousius — CHANGES_REQUESTED** [review]
Yay! Consistency! Just needs a few more tests and this LGTM @mbs-octoml

File: src/relay/ir/expr.cc (hunk 1 above)
Comment (mbs-octoml) [reply src/relay/ir/expr.cc:30]: Done.

**Comment by mbs-octoml** [comment]
Looks like #11631 and #11513 clashed, so I'm going to include the trivial fix as a rider in the PR.

**Comment by mbs-octoml** [comment]
Thanks @Mousius, pls merge when you get a chance.

**Review by Mousius — APPROVED** [review]

**Pull request merged**